docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,946 @@
+
+- [The Ethereum Virtual Machine](https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Deconstructing a Smart Contract](https://blog.openzeppelin.com/deconstructing-a-solidity-contract-part-i-introduction-832efd2d7737/)
+- [Layout of State Variables in Storage](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_storage.html)
+- [Layout in Memory](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_memory.html)
+- [Layout of Call Data](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_calldata.html)
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+### The Process
+
+We will first create a Solidity interface that our precompile will implement. Then we will use the precompile tool to autogenerate functions and fill out the rest. We're not done yet! We will then have to update a few more places within the Subnet-EVM. Some of this work involves assigning a precompile address, adding the precompile to the list of Subnet-EVM precompiles, and finally enabling the precompile. Now we can see our functions in action as we write another solidity smart contract that interacts with our precompile. Lastly, we will write some tests to make sure everything works as promised. We will also have an [official tutorial](https://github.com/ava-labs/hello-world-official-precompile-tutorial) with step by step commits you can follow to double check your work.
+
+### Prerequisites
+
+- Install Go 1.18.1 or later. Follow the instructions [here](https://golang.org/doc/install). You can verify by running `go version`
+- Clone the [Subnet-EVM](https://github.com/ava-labs/subnet-evm) repo
+- Clone [AvalancheGo](https://github.com/ava-labs/avalanchego) repo
+- Install [Avalanche Network Runner](https://docs.avax.network/subnets/network-runner)
+- Install solc by following
+
+  ```bash
+  brew tap ethereum/ethereum
+  brew install solidity
+  ```
+
+  After this you should be able to run `solc` on your command console.
+
+## Tutorial
+
+For the tutorial, we will be working in a new branch in [Subnet-EVM](https://github.com/ava-labs/subnet-evm). There will be code snippets to copy and paste in the appropriate files.
+
+We will first start off by creating the Solidity interface that we want our precompile to implement. This will be the HelloWorld Interface. It will have two simple functions, `sayHello` and `setGreeting`. These two functions will demonstrate the getting and setting respectively of a value using state access.
```

Comment:
```suggestion
We will first start off by creating the Solidity interface that we want our precompile to implement. This will be the HelloWorld Interface. It will have two simple functions, `sayHello` and `setGreeting`. These two functions will demonstrate the getting and setting respectively of a value stored in the precompile's state space.
```

Reply:
thanks

Reply:
I will fix this change with my review.